### PR TITLE
Add Chart to deploy Couchbase Operator

### PR DESCRIPTION
**Pull request by tahmmee**
Signed-off-by: Tommie McAfee <tommie@couchbase.com>

#### What this PR does / why we need it:
The Chart helps Kubernetes administrators to deploy the Couchbase Operator for deploying and managing Couchbase Clusters.  
Manual deployment of the operator requires the creation of the following resources which are automatically (and optionally) installed by this chart:

```
RESOURCES:
==> v1/ServiceAccount
NAME                                               AGE
unrealized-tarsier-couchbase-admission-controller  1h
unrealized-tarsier-couchbase-operator              1h

==> v1beta1/ClusterRole
unrealized-tarsier-couchbase-admission-controller  1h
unrealized-tarsier-couchbase-operator              1h

==> v1/ClusterRoleBinding
unrealized-tarsier-couchbase-admission-controller  1h

==> v1beta1/ClusterRoleBinding
unrealized-tarsier-couchbase-operator  1h

==> v1beta1/MutatingWebhookConfiguration
unrealized-tarsier-couchbase-admission-controller  1h

==> v1beta1/ValidatingWebhookConfiguration
unrealized-tarsier-couchbase-admission-controller  1h

==> v1/Secret
unrealized-tarsier-couchbase-admission-controller  1h

==> v1/Service
unrealized-tarsier-couchbase-admission-controller  1h

==> v1/Deployment
unrealized-tarsier-couchbase-admission-controller  1h
unrealized-tarsier-couchbase-operator              1h
```

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: incubator/couchbase-operator/.helmignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+.svn/
+# Common backup files
+*.swp
+*.bak
+*.tmp
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
```
Comment (cpanato) [inline incubator/couchbase-operator/.helmignore]: add OWNERS here as well

**Comment by tahmmee** [comment]
Hey @cpanto and @prydonius,
Let me know if any additional changes are needed here.

File: incubator/couchbase-operator/templates/admission-cluster-role-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{{- if and .Values.rbac.create  .Values.deployments.admissionController -}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "admission-controller.fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/admission-cluster-role-binding.yaml]: can you update the labels to use the new ones? if you use the latest helm when you create the project it already add the new lables, but here is for reference https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#metadata

**Comment by tahmmee** [comment]
Hey @cpanto, added the requested change, let me know if anything else needed.
Thanks!

**Comment by tahmmee** [comment]
Hi @prydonius Looks like you were requested for a review here, can you have a look?

File: incubator/couchbase-operator/templates/admission-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    helm.sh/chart: {{ include "couchbase-operator.chart" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "admission-controller.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "admission-controller.name" . }}
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/admission-deployment.yaml]: wrong label here please update to match

File: incubator/couchbase-operator/templates/admission-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  template:
+    metadata:
+      labels:
+        app: {{ template "admission-controller.name" . }}
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/admission-deployment.yaml]: here as well

File: incubator/couchbase-operator/templates/admission-service.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+metadata:
+  name: {{ template "admission-controller.service.name" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "admission-controller.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ include "couchbase-operator.chart" . }}
+spec:
+  selector:
+    app: {{ template "admission-controller.name" . }}
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/admission-service.yaml]: wrong label

File: incubator/couchbase-operator/templates/couchbase-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    helm.sh/chart: {{ include "couchbase-operator.chart" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "couchbase-operator.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "couchbase-operator.name" . }}
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/couchbase-deployment.yaml]: same here

File: incubator/couchbase-operator/templates/couchbase-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  template:
+    metadata:
+      labels:
+        app: {{ template "couchbase-operator.name" . }}
```
Comment (cpanato) [inline incubator/couchbase-operator/templates/couchbase-deployment.yaml]: here

**Review by cpanato — CHANGES_REQUESTED** [review]
also when you fix the small issues you can move this chart to the stable folder

**Comment by tahmmee** [comment]
@cpanato all set - thanks!

**Comment by maorfr** [comment]
/ok-to-test

cc @cpanato

**Review by cpanato — APPROVED** [review]
/lgtm

thanks for the work on this chart!

**Pull request merged**